### PR TITLE
Feature/peerDiscovery

### PR DESCRIPTION
**Pull request by vaishnav-ch**
DID published/registered in the network is stored in `DIDPeerTable` and can be filtered and fetched from `did_last_char` column according to the last character of DID.

### CODE REVIEW

**Review by gklps — APPROVED** [review]

File: core/wallet/did.go [hunk 1]
```diff
@@ -72,17 +73,40 @@ func (w *Wallet) IsDIDExist(did string) bool {
...
 }
 
+func (w *Wallet) AddDIDChar() error {
+	var existingDIDPeer []DIDPeerMap
+	err := w.s.Read(DIDPeerStorage, &existingDIDPeer, "did!=?", "")
+	if err != nil {
+		w.log.Error("Unable to read from DID Peer table", err)
+		return err
+	}
+	for i := 0; i < len(existingDIDPeer); i++ {
```
Comment (vyshnavsdeepak) [inline core/wallet/did.go:97]: This is to update existing data in DIDPeerStorage table,  right?

So, we need a check on do the query in a way that existing records are not updated every time

File: core/wallet/did.go [hunk 2]
```diff
@@ -72,17 +73,40 @@ func (w *Wallet) IsDIDExist(did string) bool {
...
 		return w.s.Write(DIDPeerStorage, &dm)
 	}
+
 	dm.PeerID = peerID
 	return w.s.Update(DIDPeerStorage, &dm, "did=?", did)
 }
 
+func (w *Wallet) AddDIDChar() error {
+	var existingDIDPeer []DIDPeerMap
+	err := w.s.Read(DIDPeerStorage, &existingDIDPeer, "did!=?", "")
```
[the remark this answers is not in the document]
Comment (gklps) [reply core/wallet/did.go:92]: As far as did is a primary key, will you be able to add repeative data?

File: core/peer.go [hunk 3]
```diff
@@ -56,6 +56,7 @@ func (c *Core) peerCallback(peerID string, topic string, data []byte) {
 	if err != nil || !st {
 		return
 	}
+	c.w.AddDIDChar()
```
[the remark this answers is not in the document]
Comment (vaishnav-ch) [reply core/peer.go:59]: Changed function call to `core`. Call will happen when starting core.

File: core/wallet/did.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (vaishnav-ch) [reply core/wallet/did.go:92]: Changed in such a way that only not updated DIDs will get updated. Others are not repeatedly updating.

**Review by harirubix — APPROVED** [review]

**Comment by harirubix** [comment]
@muraliens Can you merge this PR please?

File: core/wallet/did.go [hunk 4]
```diff
@@ -72,17 +73,41 @@ func (w *Wallet) IsDIDExist(did string) bool {
 }
 
 func (w *Wallet) AddDIDPeerMap(did string, peerID string) error {
```
Comment (harirubix) [inline core/wallet/did.go:75]: How do we add `did_last_char` to `DIDPeerTable`  of existing DIDs and publish to network?

**Review by harirubix — CHANGES_REQUESTED** [review]

File: core/wallet/did.go (hunk 4 above)
Comment (vaishnav-ch) [reply core/wallet/did.go:75]: Please refer the 'func (w *Wallet) AddDIDLastChar()' , this function will check for DIDs with did_last_char column as null and update them with corresponding last character.

**Review by harirubix — APPROVED** [review]

**Pull request merged**